### PR TITLE
[Improve][Connector-V2][Mongodb]allow null value fields

### PR DESCRIPTION
**Pull request by wunan1210**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/config/MongodbConfig.java [hunk 1]
```diff
@@ -104,6 +104,12 @@ public class MongodbConfig {
                     .withDescription(
                             "By utilizing flatSyncString, only one field attribute value can be set, and the field type must be a String. This operation will perform a string mapping on a single MongoDB data entry.");
 
+    public static final Option<Boolean> ALLOW_NULL =
+            Options.key("allow-null")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("allow null value when fields reading is null");
+
     // --------------------------------------------------------------------
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/config/MongodbConfig.java]: Please describe this parameter in the documentation.
https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/MongoDB.md

File: docs/en/connector-v2/source/MongoDB.md [hunk 2]
```diff
@@ -85,6 +85,7 @@ Source Options
...
 | max.time-min         | Long    | No       | 600              | This parameter is a MongoDB query option that limits the maximum execution time for query operations. The value of maxTimeMin is in Minute. If the execution time of the query exceeds the specified time limit, MongoDB will terminate the operation and return an error.                                     |
 | flat.sync-string     | Boolean | No       | true             | By utilizing flatSyncString, only one field attribute value can be set, and the field type must be a String. This operation will perform a string mapping on a single MongoDB data entry.                                                                                                                      |
+| allow-null           | Boolean | No       | false            | If set to true, exceptions will be ignored if a null value is encountered.                                                                                                                                                                                                                                     |
```
Comment (liugddx) [inline docs/en/connector-v2/source/MongoDB.md]: `allow.null` is better.

**Review by liugddx — DISMISSED** [review]
LGTM

**Comment by liugddx** [comment]
@MonsterChenzhuo PTAL

**Comment by MonsterChenzhuo** [comment]
LGTM

**Comment by MonsterChenzhuo** [comment]
Please add the corresponding e2e test.

**Comment by MonsterChenzhuo** [comment]
Example：https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-mongodb-e2e

**Comment by wunan1210** [comment]
> 



> Example：https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-mongodb-e2e

would you please tell me how to run e2e locally? error happened when using ` mvn test -pl seatunnel-e2e/seatunnel-connector-v2-e2e/connector-mongodb-e2e -am -T 1C`

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.319 s (Wall Clock)
[INFO] Finished at: 2023-06-27T09:10:49+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:testCompile (default-testCompile) on project seatunnel-config-shade: Compilation failure: Compilation failure: 
[ERROR] /home/ubuntu/seatunnel/seatunnel-config/seatunnel-config-shade/src/test/java/org/apache/seatunnel/config/JsonFormatTest.java:[20,53] error: cannot find symbol
[ERROR]   symbol:   class Config
[ERROR]   location: package org.apache.seatunnel.shade.com.typesafe.config
[ERROR] /home/ubuntu/seatunnel/seatunnel-config/seatunnel-config-shade/src/test/java/org/apache/seatunnel/config/JsonFormatTest.java:[21,53] error: cannot find symbol
```

**Comment by EricJoy2048** [comment]
> 

I think you need run `./mvnw -U -T 1C clean install -DskipTests -D"maven.test.skip"=true -D"maven.javadoc.skip"=true -D"checkstyle.skip"=true -D"license.skipAddThirdParty"` first.

**Comment by ic4y** [comment]
"==> expected: <0> but was: <1>
at org.apache.seatunnel.e2e.connector.v2.mongodb.MongodbIT.testMongodbNullValue(MongodbIT.java:149)

Please fix the e2e issue.

**Comment by EricJoy2048** [comment]
Please fix conflicts

**Comment by EricJoy2048** [comment]
Hi, Are there any latest developments regarding this PR? At present, it has been at least two weeks without any new changes

**Comment by bulolo** [comment]
any process?

**Comment by EricJoy2048** [comment]
@chl-wxp

**Comment by EricJoy2048** [comment]
@chl-wxp

**Comment by FuYouJ** [comment]
If this problem still exists, I will try to fix it

**Comment by Hisoka-X** [comment]
fixed by #6967

**Pull request closed without merging**